### PR TITLE
Extending available GPRs from 8 to 16

### PR DESCRIPTION
**Pull request by kommisar5150**
By changing a bit of code in 6 of the files, I was able to add extra registers for use without affecting the overall code. There should now be 16 general purpose registers available. However, this uses up all the binary addresses available for registers. Any new features to be added will need to extend the binary code.

### CODE REVIEW

**Comment by pg-cse** [comment]
This code update would create a couple of side effects with the language itself. 
- Creation of 2 "registers banks" (a to h then j to p)
- This forces people to remember more details about how registers are working in Capua 
This is because i register is the instruction pointer and can't be used as GPR. I am not sure what to think about this. Also the s pointer would create similar issues in the future when registers are once more extended. I believe this is why I did not extend GPR from 4 to 16 when I did the extension to 8. I think it breaks a little bit of the beauty of the architecture.

Now this got me wondering if we would be better to literally create 2 banks of registers going from 
A to G then S and A2 to G2 then S2. This might seems strange but it would benefit future development. Let's imagine for a minute that we would want to implement something similar to HyperThreading, that second bank could be used in implementing that type of functionality at which point the "only" thing missing would be a I2 type of register. 

That naming would have the following benefit:
- Not forcing people to remember the hole in the register range
- Allowing future development
- Still allow more GPRs in the case where we don't do any future development on any "HT" type of functionnalities
- A bank with a widely different naming would also allow easy calling convention (if no HT functionalities) where the "2" registers are used for parameters passing

Bottom line, I think that the naming of the registers is important in how easy an architecture is to use. I can easily see a programmer using the "2" registers for temporary values or any similar use. Giving the registers a similar name helps the programmer naturally use these registers for similar use where I think that continuing a naming from H with no I GPR then to J would not promote that type of behaviour in programmers. In fact I think they would end up not using these registers at all.

I would like to have input from our little community on this @kommisar5150 , @cannotparse @hopel-cse 

More info about HyperThreading:
https://en.wikipedia.org/wiki/Hyper-threading

**Comment by pg-cse** [comment]
I had an offline chat with @kommisar5150 we both agreed that using a "2" type of naming for registers would be better for the architecture in the long run. Therefore, I will close and refuse this pull request.
